server/aws-lsp-codewhisperer/src/language-server/netTransform/converter.ts

```diff
@@ -1,47 +1,63 @@
-    ['net8.0', 'NET_8_0'],
-    ['net7.0', 'NET_7_0'],
-    ['net6.0', 'NET_6_0'],
-    ['net5.0', 'NET_5_0'],
-    ['netcoreapp3.1', 'NET_CORE_APP_3_1'],
-    ['netcoreapp3.0', 'NET_CORE_APP_3_0'],
-    ['netcoreapp2.2', 'NET_CORE_APP_2_2'],
-    ['netcoreapp2.1', 'NET_CORE_APP_2_1'],
-    ['netcoreapp2.0', 'NET_CORE_APP_2_0'],
-    ['netcoreapp1.1', 'NET_CORE_APP_1_1'],
-    ['netcoreapp1.0', 'NET_CORE_APP_1_0'],
-    ['net481', 'NET_FRAMEWORK_V_4_8'],
-    ['net48', 'NET_FRAMEWORK_V_4_8'],
-    ['net472', 'NET_FRAMEWORK_V_4_7_2'],
-    ['net471', 'NET_FRAMEWORK_V_4_7_1'],
-    ['net47', 'NET_FRAMEWORK_V_4_7'],
-    ['net462', 'NET_FRAMEWORK_V_4_6_2'],
-    ['net461', 'NET_FRAMEWORK_V_4_6_1'],
-    ['net46', 'NET_FRAMEWORK_V_4_6'],
-    ['net452', 'NET_FRAMEWORK_V_4_5_2'],
-    ['net451', 'NET_FRAMEWORK_V_4_5_1'],
-    ['net45', 'NET_FRAMEWORK_V_4_5'],
-    ['net403', 'NET_FRAMEWORK_V_4_0'],
-    ['net40', 'NET_FRAMEWORK_V_4_0'],
     ['net35', 'NET_FRAMEWORK_V_3_5'],
-])
+    ['net40', 'NET_FRAMEWORK_V_4_0'],
+    ['net403', 'NET_FRAMEWORK_V_4_0'],
+    ['net45', 'NET_FRAMEWORK_V_4_5'],
+    ['net451', 'NET_FRAMEWORK_V_4_5_1'],
+    ['net452', 'NET_FRAMEWORK_V_4_5_2'],
+    ['net46', 'NET_FRAMEWORK_V_4_6'],
+    ['net461', 'NET_FRAMEWORK_V_4_6_1'],
+    ['net462', 'NET_FRAMEWORK_V_4_6_2'],
+    ['net47', 'NET_FRAMEWORK_V_4_7'],
+    ['net471', 'NET_FRAMEWORK_V_4_7_1'],
+    ['net472', 'NET_FRAMEWORK_V_4_7_2'],
+    ['net48', 'NET_FRAMEWORK_V_4_8'],
+    ['net481', 'NET_FRAMEWORK_V_4_8'],
+    ['netcoreapp1.0', 'NET_CORE_APP_1_0'],
+    ['netcoreapp1.1', 'NET_CORE_APP_1_1'],
+    ['netcoreapp2.0', 'NET_CORE_APP_2_0'],
+    ['netcoreapp2.1', 'NET_CORE_APP_2_1'],
+    ['netcoreapp2.2', 'NET_CORE_APP_2_2'],
+    ['netcoreapp3.0', 'NET_CORE_APP_3_0'],
+    ['netcoreapp3.1', 'NET_CORE_APP_3_1'],
+    ['net5.0', 'NET_5_0'],
+    ['net6.0', 'NET_6_0'],
+    ['net7.0', 'NET_7_0'],
+    ['net8.0', 'NET_8_0'],
+  ]);
+
+const targetFrameworkKeysArray = Array.from(targetFrameworkMap.keys());
+function getKeyIndexOfVersion( key: any) {
+  return targetFrameworkKeysArray.indexOf(key);
+}
+
+function findMinimumSourceVersion(projectMetadata : TransformProjectMetadata[], logging: Logging){
+    var selectedVersionIndex = 999;
+    projectMetadata.forEach(project => {
+        if(project.ProjectTargetFramework != ''){
+            if(targetFrameworkMap.has(project.ProjectTargetFramework)){
+                logging.log("Project version here " + project.ProjectTargetFramework);
+                selectedVersionIndex = getKeyIndexOfVersion(project.ProjectTargetFramework) < selectedVersionIndex ? getKeyIndexOfVersion(project.ProjectTargetFramework) : selectedVersionIndex;
+            }
+        }});
+    var selectedDotNetVersion = selectedVersionIndex != 999? targetFrameworkMap.get(targetFrameworkKeysArray[selectedVersionIndex]) : '';
+    logging.log("Selected version " + selectedDotNetVersion);
+    return selectedDotNetVersion;
+}
```

Comment:
Would recommend restructuring this function based on the following to make it more readable and less error prone:
1. Rename the function to more accurately describe what it is doing (e.g. `convertTargetFrameworkForTransformRequest`)
2. Avoid writing literal values in multiple places (e.g. store the value 999 in a const like `DEFAULT_FRAMEWORK_INDEX`). We may not even need this change though (see below)
3. In the forEach loop, `break` once you've found a match.
4. Use simpler logical flow like:

```typescript
var convertedTargetFramework = '';
if(project.ProjectTargetFramework != ''){
    projectMetadata.forEach(project => {
        convertedTargetFramework = ... ;
    }
}
logging.log("For project {projectName}, converted target framework from \"{initialFramework}\" to \"{convertedTargetFramework}\"");

return convertedTargetFramework;
```

Reply:
hi @jonlouie , I did not get why we do not need the above check, from list, we need to find 1 minimum source framework. There can be multiple projects in metadata like .net48, net6, net8. What we need to tag is net48 for gumby request. 
please lmk if I am missing something

Reply:
This is a nit, but would it be easier to read the loop if combined the conditions instead of having them be nested?

Reply:
@pranav-firake Thanks for the offline explanation. Per our conversation, we've agreed on the following changes:
1. Store 999 in a local `const`
2. Rename variables from `selected...` to `minimum...`

We did not discuss this offline, but I would suggest updating the logging statement to something like `"Lowest target framework found: {minimumDotNetVersion}"`

Reply:
thanks Jon,

Reply:
yes Chris, agreed, refactored to use single condition check rather than nested